gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/CommandServiceImpl.java

```diff
@@ -81,12 +86,13 @@ public List<CommandEntity> search(ExecutionContext executionContext, CommandQuer
             tags = query.getTags().stream().map(Enum::name).toArray(String[]::new);
         }
 
-        CommandCriteria criteria = new CommandCriteria.Builder()
-            .to(query.getTo())
-            .tags(tags)
-            .organizationId(executionContext.getOrganizationId())
-            .environmentId(executionContext.hasEnvironmentId() ? executionContext.getEnvironmentId() : null)
-            .build();
+        final CommandCriteria.Builder criteriaBuilder = new CommandCriteria.Builder().to(query.getTo()).tags(tags);
+        if (executionContext != null) {
+            criteriaBuilder
+                .organizationId(executionContext.getOrganizationId())
+                .environmentId(executionContext.hasEnvironmentId() ? executionContext.getEnvironmentId() : null);
+        }
+        CommandCriteria criteria = criteriaBuilder.notAckBy(query.getNotAckBy()).build();
```

Comment:
Shouldn't we had the `notAckBy` criteria only if defined? Otherwise we will search `notAckBy != null`

Reply:
as discussed, if it's null then the criteria will be built with a null field and this will not be added to the query